pkg/helm/releases.go

```diff
@@ -37,3 +43,70 @@ func MakeHelmRelease(name, version, cluster, namespace string, helmRepository ty
 		},
 	}
 }
+
+// FindReleaseInNamespace iterates through a slice of HelmReleases to find one with a given name in a given namespace, and returns it with its index.
+func FindReleaseInNamespace(existingReleases []helmv2beta1.HelmRelease, name, ns string) (*helmv2beta1.HelmRelease, int, error) {
+	for i, r := range existingReleases {
+		if r.Name == name && r.Namespace == ns {
+			return &r, i, nil
+		}
+	}
+
+	return nil, -1, nil
+}
+
+// AppendHelmReleaseToString appends a HelmRelease to a string.
+func AppendHelmReleaseToString(content string, newRelease *helmv2beta1.HelmRelease) (string, error) {
+	var sb strings.Builder
+	if content != "" {
+		sb.WriteString(content + "\n")
+	}
+
+	helmReleaseManifest, err := kyaml.Marshal(newRelease)
+	if err != nil {
+		return "", fmt.Errorf("failed to marshal HelmRelease: %w", err)
+	}
+
+	sb.WriteString("---\n" + string(helmReleaseManifest))
+
+	return sb.String(), nil
+}
+
+// SplitHelmReleaseYAML splits a manifest file that contains one or more Helm Releases that may be separated by '---'.
+func SplitHelmReleaseYAML(resources []byte) ([]helmv2beta1.HelmRelease, error) {
+	var helmReleaseList []helmv2beta1.HelmRelease
+
+	decoder := apimachinery.NewYAMLOrJSONDecoder(bytes.NewReader(resources), 100000000)
```

Comment:
lets figure out a good value for `100000000` :D

Reply:
if anyone has a good suggestion for this, it would be much appreciated :D

Reply:
https://github.com/fluxcd/flux2/blob/f6afe7f0ec7d7722fd9e561d690c22d50ce10f23/internal/utils/utils.go#L307 

lgtm? @nikimanoledaki

Reply:
Actually that's a frigging LARGE buffer size. :D Usually it's either 4096 or 2048 which is perfectly enough. :) I opts for using what Jake linked in. :)

Reply:
Thanks yep I had changed this to `2048`, was waiting for more feedback. :)